x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_view_logic.ts

```diff
@@ -80,5 +96,24 @@ export const EngineViewLogic = kea<MakeLogicType<EngineViewValues, EngineViewAct
         return status === Status.IDLE || (!data && status === Status.LOADING);
       },
     ],
+    isLoadingEngineSchema: [
+      () => [selectors.fetchEngineSchemaApiStatus],
+      (status: EngineViewValues['fetchEngineSchemaApiStatus']) =>
+        [Status.LOADING, Status.IDLE].includes(status),
+    ],
+    schemaFields: [
+      () => [selectors.engineSchemaData],
+      (data) =>
+        data?.field_capabilities?.fields
+          ? Object.entries(data?.field_capabilities?.fields)
+              .map(([name]) =>
+                Object.entries({
+                  field_name: name,
+                  field_type: [...Object.keys(data?.field_capabilities?.fields[name])],
+                })
+              )
+              .map((fields) => Object.fromEntries(fields))
+          : [],
```

Comment:
since `Object.entries(data?.field_capabilities?.fields)` is in the ternary you don't need the `?.` you can just access this via `data.field_capabilities.fields` since you've already checked that fields was not undefined

But you could also remove the ternary completely by using nullish coalescing to default to an empty object when there is no data or fields, like this:
```suggestion
        Object.entries(data?.field_capabilities?.fields ?? {})
              .map(([name]) =>
                Object.entries({
                  field_name: name,
                  field_type: [...Object.keys(data?.field_capabilities?.fields[name])],
                })
              )
              .map((fields) => Object.fromEntries(fields)),
```